browser_patches/webkit/patches/bootstrap.diff

```diff
@@ -5891,6 +6262,400 @@ index 44737686187a06a92c408ea60b63a48ac8481334..c754a763688b52e7ddd47493296ef9b0
++}
++
++bool DragData::containsURL(FilenameConversionPolicy filenamePolicy) const
++{
++    return !asURL(filenamePolicy).isEmpty();
++}
++
++String DragData::asURL(FilenameConversionPolicy filenamePolicy, String* title) const
++{
++    if (!m_platformDragData->hasURL())
++        return String();
++    if (filenamePolicy != ConvertFilenames) {
++        if (m_platformDragData->url().isLocalFile())
++            return { };
++    }
++
++    if (title)
++        *title = m_platformDragData->urlLabel();
++    return m_platformDragData->url().string();
++}
++
++}
+diff --git a/Source/WebCore/platform/wpe/DragImageWPE.cpp b/Source/WebCore/platform/wpe/DragImageWPE.cpp
+new file mode 100644
+index 0000000000000000000000000000000000000000..a73488553e0e93e34535e64c8892b76d47ad7ab2
+--- /dev/null
++++ b/Source/WebCore/platform/wpe/DragImageWPE.cpp
+@@ -0,0 +1,67 @@
++/*
++ *  Copyright (C) 2010,2017 Igalia S.L.
++ *
++ *  This library is free software; you can redistribute it and/or
++ *  modify it under the terms of the GNU Lesser General Public
++ *  License as published by the Free Software Foundation; either
++ *  version 2 of the License, or (at your option) any later version.
++ *
++ *  This library is distributed in the hope that it will be useful,
++ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
++ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
++ *  Lesser General Public License for more details.
++ *
++ *  You should have received a copy of the GNU Lesser General Public
++ *  License along with this library; if not, write to the Free Software
++ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
++ */
++
++#include "DragImage.h"
++#include "NotImplemented.h"
++
++#include "Image.h"
++
++namespace WebCore {
++
++IntSize dragImageSize(DragImageRef)
++{
++    notImplemented();
++    return { 0, 0 };
++}
++
++void deleteDragImage(DragImageRef)
++{
++    notImplemented();
++}
++
++DragImageRef scaleDragImage(DragImageRef, FloatSize)
++{
++    notImplemented();
++    return nullptr;
++}
++
++DragImageRef dissolveDragImageToFraction(DragImageRef image, float)
++{
++    notImplemented();
++    return image;
++}
++
++DragImageRef createDragImageFromImage(Image* image, ImageOrientation)
++{
++    return image->nativeImageForCurrentFrame();
++}
++
++
++DragImageRef createDragImageIconForCachedImageFilename(const String&)
++{
++    notImplemented();
++    return nullptr;
++}
++
++DragImageRef createDragImageForLink(Element&, URL&, const String&, TextIndicatorData&, FontRenderingMode, float)
++{
++    notImplemented();
++    return nullptr;
++}
++
++}
+diff --git a/Source/WebCore/platform/wpe/SelectionData.cpp b/Source/WebCore/platform/wpe/SelectionData.cpp
+new file mode 100644
+index 0000000000000000000000000000000000000000..9f181fdfe507ad5b7a47b5c58295cf4f2725e7d8
+--- /dev/null
++++ b/Source/WebCore/platform/wpe/SelectionData.cpp
```

Comment:
Does this differ from gtk/SelectionData? If not why not add gtk/SelectionData to WPE CMake files?

Reply:
Talked about it offline. Copying the data is unfortunate, but sharing the files between gtk and wpe requires changing the build system in ways that might be harder to merge.